diff --git a/flutter/bin/internal/shared.sh b/flutter/bin/internal/shared.sh
index 2a47642b16..2a4d41a8b0 100644
--- a/flutter/bin/internal/shared.sh
+++ b/flutter/bin/internal/shared.sh
@@ -157,7 +157,7 @@ function upgrade_flutter () (
       export PUB_SUMMARY_ONLY=1
     fi
     export PUB_ENVIRONMENT="$PUB_ENVIRONMENT:flutter_install"
-    pub_upgrade_with_retry
+    #pub_upgrade_with_retry
 
     # Move the old snapshot - we can't just overwrite it as the VM might currently have it
     # memory mapped (e.g. on flutter upgrade). For downloading a new dart sdk the folder is moved,
